JPEG - Photoshop 8BIM support

janousek opened the pull request:
Some JPEG images exported from Adobe Photoshop does not contains APP0 or APP1 segment, but contains APP13 segment. These JPEG can be also processed without any problem.


Example JPEG:
![hlavicka-webove-vstupenky](https://github.com/empira/PDFsharp/assets/304745/933029c6-6cb9-4276-baf6-255d16653261)

MLaukala commented:
ScanNextToken() will usually crash if it finds itself inside of a stream due to unexpected characters. Currently, my added function IsValidXref() uses ScanNextToken() to determine if the next token is an 'xref' token. This is used to determine if a fix needs to be made. However, if startxref is wrong, the _lexer.Position could be inside of a stream which will more than likely cause an exception.

janousek committed:
JPEG - Photoshop 8BIM support

Some JPEG images exported from Adobe Photoshop does not contains APP0 or APP1 segment, but contains APP13 segment. These JPEG can be also processed without any problem.
